src/configuration/vimrc.ts

```diff
@@ -18,6 +18,60 @@ export class VimrcImpl {
     return this._vimrcPath;
   }
 
+  private static readonly SOURCE_REG_REX = /^(source)\s+(.+)/i;
+
+  private static buildSource(line: string) {
+    if (line.trimLeft().startsWith('"')) {
+      return;
+    }
+
+    const matches = VimrcImpl.SOURCE_REG_REX.exec(line);
+    if (!matches || matches.length < 2) {
+      return undefined;
+    }
+
+    const sourceKeyword = matches[1];
+    const filePath = matches[2];
+
+    return VimrcImpl.expandHome(filePath);
+  }
+
+  private static async loadConfig(config: IConfiguration, configPath: string) {
+    try {
+      const vscodeCommands = await vscode.commands.getCommands();
+      const lines = (await fs.readFileAsync(configPath, 'utf8')).split(/\r?\n/);
+      for (const line of lines) {
+        const source = this.buildSource(line);
+        if (source) {
+          if (!(await fs.existsAsync(source))) {
+            console.warn(`Unable to find "${source}" file for configuration.`);
+            continue;
+          }
+          console.log(`Loading "${source}" file for configuration.`);
+          VimrcImpl.loadConfig(config, source);
```

Comment:
Is it a problem that we don't `await` this? I think if two sourced .vimrcs have overlapping remaps, it would make the one that is effective non-deterministic, though probably that's not a big deal.

Reply:
I think `await`ing on this is probably a good idea, don't want it to be non-deterministic!